### ISSUE TITLE
Add support for texture scale + offset

### ISSUE THREAD

**Issue by cpetry** [Siccity/GLTFUtility#66]
Hi there,

when exporting materials from blender, there are texture information about scale and offset which are not yet applied on unity import.
See example here:
```JSON
"materials" : [
        {
            "doubleSided" : true,
            "name" : "someMaterial",
            "normalTexture" : {
                "extensions" : {
                    "KHR_texture_transform" : {
                        "offset" : [
                            0,
                            -3
                        ],
                        "scale" : [
                            4,
                            4
                        ]
                    }
                },
                "index" : 0,
                "texCoord" : 0
            },
     }
]
```

**Comment by Siccity** [comment]
Does this have something to do with atlassing? Looks like something that could be easily supported

**Comment by Siccity** [comment]
And can you provide a sample model

**Comment by cpetry** [comment]
I created a simple checkerCube with scale 4 in blender (zipped as github doesn't like gltf)
![image](https://user-images.githubusercontent.com/3867700/75988906-97cb4880-5ef2-11ea-975c-72604bd0f88a.png)
[checkerCube.zip](https://github.com/Siccity/GLTFUtility/files/4293286/checkerCube.zip)

Windows Paint3D loads it correctly like this:
![image](https://user-images.githubusercontent.com/3867700/75988684-402cdd00-5ef2-11ea-836c-5bc864ac3e4c.png)

With GLTFUtility it comes in like this:
![image](https://user-images.githubusercontent.com/3867700/75988859-85510f00-5ef2-11ea-807a-cfece007877c.png)

**Comment by cpetry** [comment]
It seems in gltf you can set different scale/offset settings for each texture (diffuse, normal, specular...) but in unity you can only set "tiling".
Bet this could be an additional problem

**Comment by Siccity** [comment]
What you are dealing with here is not part of the official gltf specification. KHR_texture_transform is an _extension_, and as such it is not required to be supported. I'm actually not entirely sure why this extension exists. It seems like the exporter could just as easily apply this transformation to the uv coords as I could.

**Comment by cpetry** [comment]
Well it does work when scaling the UV of the cube.
But I would need it as "tiling" inside unity, as I only want to import materials.

**Comment by cpetry** [comment]
The other option would be to increase texture resolution.... but that I don't like (file size)

**Comment by prajwalshettydev** [comment]
+1 for supporting "KHR_texture_transform"  extension. Dealing with the scale problem now.

**Comment by cpetry** [comment]
I wrecked my head around an alternative value that I can use to transfer the scale information. Sadly nothing comes to my mind.
Can you somehow give access to any information given? Like a dictionary with all values in raw?

**Comment by Siccity** [comment]
After looking into this it seems more 'correct' to apply the transformations to the material, not the UVs. This would introduce higher shader complexity, and thus lower performance of the builtin shaders.

**Comment by cpetry** [comment]
You're right. But is it that much slower calculating some uv coords before lookup? I always thought as long as there is no branching involved (i.e. bool decisions) performance isn't that big a deal. Multiplication and a sum for the offset am I right?

**Comment by Siccity** [comment]
Problem is it's *per texture*. Offsetting the UV means all textures Will have to share the same offset, which might not always be the case. The only solution I see is using individual TextureScaleOffset properties with all material textures, and also a rotation value. All this extra shader stuff will have a negative performance impact for all users.

**Comment by cpetry** [comment]
I am actually currently playing aroung with my own shaders myself (some seeThrough stuff).
Removing the tag `[NoScaleOffset]` and applying tiling and offset via C# should do the trick, I assume.
In case you have performance doubts we could do some profiling/benchmarking.

**Comment by jysung** [comment]
Did you have any luck? I would also like to have  `KHR_texture_transform` support

**Comment by HawkenKing** [comment]
I've had this exact same problem with a-frame, I think it's not a problem with GLTFUtility but with GLTF it's self. Tiling is ignored somehow

**Comment by cpetry** [comment]
Made a PullRequest on this.
Please test/check and comment!

**Comment by Siccity** [comment]
Merged 82b85fd8fcc6afd62b00d4045f54c08c9298cd0f

**Comment by diegodelarocha** [comment]
Does this have texture animation support? 😍 I tried, testing this, but didn't have luck with it.

**Comment by Siccity** [comment]
Texture animation? Is it supposed to?

**Comment by diegodelarocha** [comment]
It's a bit buggy on Blender, but you can set animation on the Texture Coordinates of X and Y axis, and it would be awesome if the keyframes that you set up in Blender, get exported as animation controlling the UVOffset of a material, this way you can do sprite animation on geometry using a texture pass. This would be super helpful to achieve effects like character blinks a la Zelda: The Wind Waker style 💜

**Comment by diegodelarocha** [comment]
This video does a better job at explaining what I mean 
https://www.youtube.com/watch?v=iVu2ms207UE

On the video the animation is controlled via script, but a lot of times as animators, we want to control that through animation, which is very helpful if we can preview from Blender 🤓

**Comment by Siccity** [comment]
Is this part of the spec?

**Comment by diegodelarocha** [comment]
You mean the GLTF spec? 🤓 Sorry, my knowledge is a bit limited, I'm just a romantic animator hoping to get features in, but my technical knowledge is a bit limited. I tried checking and it might not be? 😭 Maybe I couldn't find it though :/

I did a test, on blender, where an object is driving the the X and Y axis of the Mapping Node, which lets us preview this in Blender, maybe there's a way to hook that driver object, into the `UVOffset` property of Unity?

Here's the driver object:
<img width="864" alt="driverObject" src="https://user-images.githubusercontent.com/20521814/79781782-e0985e80-82f2-11ea-8daf-f39924c5fb79.png">

And here's the driven texture inheriting the same values that the driver object has:
<img width="771" alt="drivenTexture" src="https://user-images.githubusercontent.com/20521814/79781765-dbd3aa80-82f2-11ea-94e5-225e23072351.png">

Here's a zip with a blender file and a GLTF file of the picture's test:
[UVTilesAndAndSets_004_objectDriven.zip](https://github.com/Siccity/GLTFUtility/files/4505083/UVTilesAndAndSets_004_objectDriven.zip)
